sys/net/application_layer/uhcp/uhcp.c

```diff
@@ -98,8 +98,8 @@ void uhcp_handle_push(uhcp_push_t *req, uint8_t *src, uint16_t port, uhcp_iface_
     char prefix_str[INET6_ADDRSTRLEN];
     inet_ntop(AF_INET6, src, addr_str, INET6_ADDRSTRLEN);
     uint8_t prefix[IN6ADDRSZ] = { 0 };
-    if (req->prefix_len == 0) {
-        LOG_ERROR("uhcp_handle_push(): prefix length 0\n");
+    if ((req->prefix_len == 0) || (req->prefix_len > (IN6ADDRSZ << 3))) {
```

Comment:
I chose this notation to be in line with the notation below: `prefix` is of size `IN6ADDRSZ`, there are `prefix_bytes` bytes copied to `prefix` which is generated by bit-shifting `req->prefix_len` (+ 7 to round up) three bits to the right. So `req->prefix_len` must not be larger than `INADDRSZ` bit-shifted three bits to the left. Or to state it less complicated, `req->prefix_len` must be greater or equal to `INADDRSZ * 8` (bits), or 128...